source/upgrade.txt

```diff
@@ -0,0 +1,173 @@
+.. _node-upgrade-driver:
+
+=======================
+Upgrade Driver Versions
+=======================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+On this page, you can learn about any changes you might need to make to
+your application to upgrade your driver to a new version without loss of
+functionality.
+
+Before you upgrade, perform the following actions:
+
+- Ensure the new driver version is compatible with the {+mdb-server+} version
+  your application connects to and the version of Node.js that your
+  application runs on. See the :ref:`<node-compatibility>`
+  page for this information.
+- Address any breaking changes between the current version of the driver
+  your application uses and your planned upgrade version in the
+  :ref:`<node-breaking-changes>` section of this guide. To learn
+  more about the {+mdb-server+} release compatibility changes, see the 
+  :ref:`<node-server-support-changes>` section.
+
+.. tip::
+
+   You can minimize the amount of changes that you need to make to your
+   application when upgrading driver versions by using the
+   :ref:`{+stable-api+} <nodejs-stable-api>`.
+
+.. _node-breaking-changes:
+
+Breaking Changes
+----------------
+
+A breaking change is a modification in a convention or behavior in
+a specific version of the driver that may prevent your application from
+working as expected.
+
+The breaking changes in this section are categorized by the major
+version releases that introduced them. When upgrading driver versions,
+address all the breaking changes between the current and upgrade
+versions. For example, if you are upgrading the driver from v3.x to
+v5.x, address all breaking changes listed under v4.x and v5.x.
+
+.. _node-breaking-changes-v5.x:
+
+Version 5.x Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Driver versions 5.x are not compatible with Node.js
+  v12 or earlier. If you want to use this version of the driver, You must
+  use Node.js v14.20.1 or greater.
+
+- The driver removes support for callbacks in favor of a promise-based API.
+  The following list provides some strategies for callback users to adopt this
+  version:
+
+  - Migrate to the promise-based API (recommended)
+  - Use the promise-based API and ``util.callbackify``
+  - Add ``mongodb-legacy`` to continue using callbacks
+
+For more information about these strategies, see
+`the v5.0 changelog <https://github.com/mongodb/node-mongodb-native/blob/main/etc/notes/CHANGES_5.0.0.md#optional-callback-support-migrated-to-mongodb-legacy>`__.
```

Comment:
Indent this line so that it falls in line with it's list item (not sure if the suggestion below will format correcly)

```suggestion
  - Add ``mongodb-legacy`` to continue using callbacks

  For more information about these strategies, see
  `the v5.0 changelog <https://github.com/mongodb/node-mongodb-native/blob/main/etc/notes/CHANGES_5.0.0.md#optional-callback-support-migrated-to-mongodb-legacy>`__.
```